sklearn/metrics/_classification.py

```diff
@@ -2647,10 +2656,8 @@ def log_loss(
     ...          [[.1, .9], [.9, .1], [.8, .2], [.35, .65]])
     0.21616...
     """
-    y_pred = check_array(
-        y_pred, ensure_2d=False, dtype=[np.float64, np.float32, np.float16]
-    )
```

Comment:
Parameter validation does not make data validation obsolete. It just checks that the type y_pred is appropriate but does not check what check_array does, i.e its dtype, shape, content... We need to keep this check.